inc/models/class-profile.php

```diff
@@ -19,19 +19,31 @@
  *
  * Dynamic properties.
  *
+ * @property int    $byline_id    Term ID for the profile.
+ * @property string $description  Profile description.
+ * @property string $display_name Display name for the profile.
+ * @property string $link         Profile permalink.
  * @property int    $post_id      Post ID for the profile.
  * @property int    $term_id      Term ID for the profile.
- * @property string $display_name Display name for the profile.
+ * @property string $user_nicename User nicename.
  * @property string $user_url     User url.
- * @property string $link         Profile permalink.
  */
 class Profile {
 	/**
 	 * Profile post object.
 	 *
 	 * @var WP_Post
 	 */
-	public $post;
+	public WP_Post $post;
```

Comment:
This property type isn't enforced in the constructor or its docs. If the thinking is that it's safe to add the type because all the static constructors route through `::get_by_post()` anyway, I think it'd be worth concretizing that now by adding the type to the constructor and updating the docs.